source/uwp/Renderer/lib/TextHelpers.cpp

```diff
@@ -177,7 +177,8 @@ HRESULT SetXamlInlines(_In_ ABI::AdaptiveNamespace::IAdaptiveTextElement* adapti
     if (markdownParser.HasHtmlTags())
     {
         HString htmlHString;
-        UTF8ToHString(htmlString, htmlHString.GetAddressOf());
+        // wrap html in <root></root> in case there's more than one toplevel element.
```

Comment:
>// wrap html in <root></root> in case there's more than one toplevel element. [](start = 8, length = 77)

Do we want to do this in the markdown parser itself? (or does that break other platforms?)

Also, out of curiosity, did you ever hunt down whether this issue had always existed or was new?

Reply:
I think it might break other platforms -- particularly since I think Android (and possibly iOS) use HTML parsers to handle markdown output.

AFAIK this behavior has always existed. I filed #3337 to describe the behavior using latin chars...

---
In reply to: [314841203](https://github.com/microsoft/AdaptiveCards/pull/3342#discussion_r314841203) [](ancestors = 314841203)